photon-client/src/main/scala/com/linkedin/photon/ml/util/IOUtils.scala

```diff
@@ -365,4 +367,41 @@ object IOUtils {
 
     builder.mkString
   }
+
+  /**
+   * Save GAME per-group evaluation in terms of (random effect id, evaluation value) to HDFS.
+   *
+   * @param sparkSession The Spark session
+   * @param outputDir The directory in HDFS where to save the evaluation
+   * @param evaluationResultsOpt An Option of evaluation results
+   * @param logger The logger instance for the application
+   */
+  def saveGameEvaluationToHDFS(
+    sparkSession: SparkSession,
+    outputDir: Path,
+    evaluationResultsOpt: Option[EvaluationResults],
+    logger: PhotonLogger): Unit = {
+
+    evaluationResultsOpt match {
+      case Some(evaluationResults) => evaluationResults.evaluations.foreach {
+        case (evaluatorType, (_, Some(groupEval))) => {
+          val evaluatorName = evaluatorType.name.split(MultiEvaluatorType.shardedEvaluatorIdNameSplitter)
+          logger.debug(s"Save per-group evaluation of ${evaluatorName(0)} on ${evaluatorName(1)}")
+          val evalOutputDir = new Path(new Path(outputDir, evaluatorName(0)), evaluatorName(1))
+          sparkSession
+            .createDataFrame(groupEval)
```

Comment:
Do we need to define some pre-specified schema?

Reply:
The schema is determined by the arguments in the following `toDF`.